CIPs/CIP-79/cip-79.md

```diff
@@ -0,0 +1,282 @@
+```
+
+### 3IDv0
+
+The method specific identifier for 3IDv0 is a [CID](https://github.com/multiformats/cid) as produced by the [IPLD](https://github.com/ipld/specs) codec [dag-cbor](https://github.com/ipld/specs/blob/master/block-layer/codecs/dag-cbor.md).
+
+```
+3idv0 = "did:3:<cid>"
+```
+
+#### Example
+
+```sh
+did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi
+```
+
+## CRUD Operation Definitions
+
+In this section the CRUD operations for a 3ID DID are defined.
+
+### Create
+
+A `3` DID is created by simply creating a Ceramic [`tile`](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-8/CIP-8.md) document. The [`tile`](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-8/CIP-8.md) document type takes a DID as the *controller*, and it's recommended that a `did:key` is used. The *controller* is the DID which is allowed to update the document. The *family* of the document is set to `3id`, and the *deterministic* flag is set to `true`.
+
+Now the content of the document should consist of a JSON object with one property *publicKeys*. These public keys will be allowed to sign messages on behalf of the 3ID, or decrypt messages encrypted to the 3ID. The value of this property should be an object where the value is a [multicodec](https://github.com/multiformats/multicodec/) + multibase(*base58btc*) encoded public key, the key for any given key should be the last *15* characters of the encoded public key.
+
+The 3ID DID method supports any type of public key that can be encoded using multicodec which can easily be extended to support quantum resistant signature and encryption schemes in the future.
+
+#### Example
+
+Below you can see an example of how to create a 3ID using the Ceramic javascript api.
+
+```js
+const doc = await ceramic.createDocument({
+  metadata: {
+    controllers: ['did:key:zQ3shQNcackrTByiYaPGso1Nt7b6r1gSMg4XXBmavzvTMqX1h'], // secp256k1 did:key
+    family: '3id'
+  },
+  content: {
+    publicKeys: {
+      "XQCT2xJHsdY6iJH": "z6LSqKWh3XQ7AfsJuE2KR23cozEut8D5CXQCT2xJHsdY6iJH",  // secp256k1 public key
+      "yh27jTt7Ny2Pwdy": "zQ3shrMGEKAjUTMmvDkcZ7Y3x9XnVjTH3myh27jTt7Ny2Pwdy", // x25519 public key
+    }
+  },
+  deterministic: true
+})
+
+const didString = `did:3:${doc.id}`
+```
+
+### Read/Verify
+
+Resolving a 3ID is quite straight forward. It is done by taking the DocID from the method specific identifier, looking up the referred document using Ceramic, and converting the content of the Ceramic document into a DID document.
+
+#### Loading 3IDv1 document
+
+To load a 3IDv1 document simply take the DocID from the method specific identifier and load the document from ceramic.
+
+#### Loading 3IDv0 document
+
+To load a 3IDv0 document, first take the CID from the method specific identifier and load the corresponding object from the IPFS network. The resolved object will contain a `publicKey` property which has an array of public key objects (see [Appendix A](#appendix-a)).
```

Comment:
somewhat covered by `3IDv0 genesis` section below, but maybe note what happens if no ceramic doc is resolved in the steps here

Reply:
Well, the genesis commit will be resolved regardless since it's deterministically generated. Will see if I can improve it a bit!